.github/workflows/integration-test-iOS16.yaml

```diff
@@ -12,7 +12,7 @@ on:
 
 jobs:
   check:
-    runs-on: macos-10.15
+    runs-on: macos-latest
```

Comment:
I know that @QuintinWillison [mentioned on Slack](https://ably-real-time.slack.com/archives/CSQEKCE81/p1669707060644419?thread_ts=1669666028.231469&cid=CSQEKCE81) that he'd prefer to use `latest` instead of locking a version. I'd be curious to understand the reason. Personally, I'd prefer that we remain in control of when our CI environment changes (but we _should_ be making these changes more frequently than we currently do, to keep up as new runner versions are released).

Also, as mentioned in another comment here, the macOS version chosen for the runners also affects the macOS version which we use for testing the macOS version of the SDK, which I think should be locked for consistency with how we choose iOS and tvOS versions.

Reply:
My preference is just a personal opinion, in the absence of a better plan. In my experience we've generally seen less ongoing maintenance overhead and friction by sticking with _convention over configuration_, thus not explicitly overriding defaults unless we have a strong reason to. In theory, our builds should always be able to execute and succeed on the latest tooling, as well as more recent versions of the tooling older than that (macOS test target is a separate concern, albeit related due to the way Apple's tooling works)

If we are going to proceed with a policy of pinning down the version of macOS being used by our CI jobs then I would like to see documentation as to why we're selecting a specific version now, plus a strategy for ensuring we proactively upgrade it from time to time on an appropriate cadence presumably aligned with Apple (macOS) or GitHub (`macos-` runners) releases.

By opting for `macos-latest`, sticking with convention, the counter argument is that we're automatically gaining a "fail early" behaviour in CI when GitHub update the runners to newer versions of the OS. 🤔 

I think there's pros and cons of both approaches.

Reply:
>  In theory, our builds should always be able to execute and succeed on the latest tooling

Definitely. I think my concern is that choosing `macos-latest` doesn't allow us to _choose how we prioritise_ the work needed to make sure we satisfy this requirement. If CI starts failing because of a runner version change, we have no choice but to immediately address that failure, even if there are more urgent things we would prefer to be working on. This is because CI failure blocks us from doing any other work.

> plus a strategy for ensuring we proactively upgrade it from time to time

That's very reasonable. I don't have a good idea for exactly what that would look like, other than documenting a process that expects developers to raise an issue when a new version of Xcode is released or something along those lines. (Unless we wanted to be fancier and have some sort of automation that checks available runner versions.)

> macOS test target is a separate concern

Do you have any thoughts on how we should address that one? Would you prefer that we move to running macOS tests on the latest version available?

Reply:
Thanks, @lawrence-forooghian. I can't disagree with anything you've stated above, however right now I feel that we should leave this at `macos-latest` which will solve the issue being addressed by @maratal and allow this pull request to merge and mitigate the immediate term risk (that being the `macos-10.15` image disappearing from under us!). Longer term I've created https://github.com/ably/ably-cocoa/issues/1544 which is probably the most appropriate venue to continue this conversation on, plus of course the issue under which we can work to remove `macos-latest` from the mix.

Reply:
Cool, thanks for creating that issue @QuintinWillison. In that case, @maratal, I think it's just a case of addressing [this comment](https://github.com/ably/ably-cocoa/pull/1543#discussion_r1034979446).